src/react/utils/useHydrateAtoms.ts

```diff
@@ -36,7 +38,12 @@ export function useHydrateAtoms<T extends Iterable<AtomTuple>>(
 
   const hydratedSet = getHydratedSet(store)
   for (const [atom, value] of values) {
-    if (!hydratedSet.has(atom)) {
+    const isHydratedAtom = hydratedSet.has(atom)
+    if (!isHydratedAtom || options?.forceHydrate) {
+      if (options?.forceHydrate && isHydratedAtom) {
+        hydratedSet.delete(atom)
+      }
+
```

Comment:
Can it be as simple as this?
```suggestion
    if (!hydratedSet.has(atom) || options?.dangerouslyForceHydrate) {
```